lib/processor.js

```diff
@@ -140,7 +274,25 @@ function adjustBlock(block) {
             column: message.column + block.position.indent[lineInCode - 1] - 1
         };
 
-        return assign({}, message, out);
+        var adjustedFix = {};
+        if (message.fix) {
+            adjustedFix.fix = {
+                range: message.fix.range.map(function(range) {
+                    // Advance through the block's range map to find the last
+                    // matching range.
+                    var i = 0;
+                    while (i < block.rangeMap.length && block.rangeMap[i] < range) {
+                        i += 2;
+                    }
```

Comment:
See my comment above.

Then here we can just simply iterate through the array, and be more explicit. What do you think?